test/integration/cluster/cluster_test.go

```diff
@@ -0,0 +1,139 @@
+	ClusterNetwork: clusterv1alpha1.ClusterNetworkingConfig{
+		ServiceDomain: "mydomain.com",
+		Services: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"10.96.0.0/12"},
+		},
+		Pods: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"192.168.0.0/16"},
+		},
+	},
+}
+
+// Constructs a function for async event retrieval
+// amenable to be used with Gomega's Eventually assertion
+func checkEvents(c chan *corev1.Event) func() *corev1.Event {
+	return func() *corev1.Event {
+		select {
+		case e := <-c:
+			return e
+		default:
+			return nil
+		}
+	}
+}
+
+const TIMEOUT = 10 * time.Second
+const RETRY = 100 * time.Millisecond
+
+func TestCluster(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cluster-Controller")
+}
+
+var _ = Describe("Cluster-Controller", func() {
+	var clusterapi client.ClusterInterface
+	var client *kubernetes.Clientset
+	var stopper chan struct{}
+	var informer cache.SharedIndexInformer
+	var testNamespace string
+
+	BeforeEach(func() {
+		// Load configuration
+		kubeconfig := os.Getenv("KUBECONFIG")
+		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Create kubernetes client
+		client, err = kubernetes.NewForConfig(config)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Create namespace for test
+		ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "clusterapi-test-"}}
+		ns, err = client.Core().Namespaces().Create(ns)
+		Expect(err).ShouldNot(HaveOccurred())
+		testNamespace = ns.ObjectMeta.Name
+
+		// Create  informer for events in the namespace
+		factory := informers.NewSharedInformerFactoryWithOptions(client, 0, informers.WithNamespace(testNamespace))
+		informer = factory.Core().V1().Events().Informer()
+		stopper = make(chan struct{})
+
+		// Create clusterapi client
+		cs, err := clientset.NewForConfig(config)
+		Expect(err).ShouldNot(HaveOccurred())
+		clusterapi = cs.ClusterV1alpha1().Clusters(testNamespace)
+	})
+
+	AfterEach(func() {
+		close(stopper)
+		client.Core().Namespaces().Delete(testNamespace, &metav1.DeleteOptions{})
+	})
+
+	Describe("Create Cluster", func() {
+		It("Should trigger an event", func() {
+			// Register handler for cluster events
+			events := make(chan *corev1.Event, 0)
+			informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					e := obj.(*corev1.Event)
+					if e.InvolvedObject.Kind == "Cluster" {
+						events <- e
+					}
+				},
+			})
+			go informer.Run(stopper)
+			Expect(cache.WaitForCacheSync(stopper, informer.HasSynced)).To(BeTrue())
+
+			// Create Cluster
+			cluster := &clusterv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "cluster-",
+					Namespace:    testNamespace,
+				},
+				Spec: *clusterSpec.DeepCopy(),
+			}
+			_, err := clusterapi.Create(cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Eventually(checkEvents(events), TIMEOUT, RETRY).ShouldNot(BeNil())
```

Comment:
We alternatively could consider either using this model (https://onsi.github.io/ginkgo/#asynchronous-tests) or making sure we set a reasonable `--timeout` value when running `ginkgo` (the default is 24hr) or `-timeout` for `go test` (the default is 10m).

Reply:
+1 for ginkgo asynchronous-test.

Reply:
@ncdc @figo I already using asynchronous test for receiving the events in line 135 with `Eventually`.  What other asynchronous check are you suggesting?

Reply:
@ncdc @figo The test are already asynchronous. The informer notifies the test of events by means of channels and the test is done using `gomega`'s `Eventually` assertion.  Not sure what specific changes you expect.

Reply:
It's not necessary, but I was trying to avoid a polling loop (which is what `Eventually` does). Something like this:

```go
It("should trigger an event", func(done Done) {
  // ...
  Expect(<-events).ShouldNot(BeNil())
  close(done)
}, TIMEOUT)
```

Reply:
@ncdc this was the patter of my initial code, but ti was suggested (IIRC by @figo) to avoid the blocking `Expect(<-events)` and I agree that doing it non-blocking channel receive was better. Therefore I moved to use `Eventually` and a non-blocking receive, which requieres some sort of polling.

Reply:
If you use the code pattern I wrote above, it allows you to have a timeout with needing to poll.

Either approach should be fine. I was suggesting something that I would prefer (not polling + you don't have to write a `checkEvents` function), but if you prefer this way, 👍 and let's get this merged.

Reply:
I see, I agree this way is simpler.